src/WebViewTypes.ts

```diff
@@ -721,6 +721,25 @@ export interface IOSWebViewProps extends WebViewSharedProps {
    * `selectedText`: the text selected on the document
    */
   onCustomMenuSelection?: (event: WebViewEvent) => void;
+
+  /**
+   * By default, if this is undefined or false, the native WebView will get released when
+   * the React component unmounts.
+   * 
+   * When this is true, the native WebView will not get released when the React component
+   * unmounts. When a React component remounts, it can use a previous native WebView instance
+   * by using the same webViewKey prop that the previous React component used.
+   * 
+   * It's important to call `release` on the React WebView with the corresponding webViewKey
+   * when the native WebView is no longer needed.
+   */
+  keepWebViewInstanceAfterUnmount?: boolean;
+
+  /**
+   * When keepWebViewInstanceAfterUnmount is true, if two React components use the same
+   * key for the WebView, they will use the same native WebView instance.
+   */
+  webViewKey?: string;
```

Comment:
What happens if `keepWebViewInstanceAfterUnmount` is `true` but `webViewKey` is `undefined`?

Reply:
As currently written, then in that scenario, the `WKWebView` will have the default behavior: its lifecycle will be roughly one to one with the lifecycle of the corresponding react component.

`keepWebViewInstanceAfterUnmount` needs to be true and `webViewKey` must be non-nullish for the `WKWebView` to stay alive after its corresponding react component unmounts. I wrote it this way to make the behavior more explicit. It's theoretically more flexible because then we can use the `webViewKey` for other cases too if necessary. I haven't actually thought of concrete examples that would need just the `webViewKey` though, so I'm open to changing this if that's simpler.

An alternative option is to remove `keepWebViewInstanceAfterUnmount` and then assume that if `webViewKey` is non-nullish, then we should keep the `WKWebView` alive after react-component-unmount.

Reply:
The alternative option is what I was originally thinking, but being explicit isn't a bad idea as long as it's documented that you need to use both.

The other use of `webViewKey` that I've been thinking of (and mentioned in #2640) would be to leverage this for supporting popup windows.... e.g., if web content calls `window.open`, the native "create window" handler could potentially create a new `WKWebView` in the dictionary and raise an event with the (generated?) key, which the app could then use (in RN code) to render a new WebView with that key. _Theoretically_, this could mean that the two WebViews would be connected as they would be in a browser.

I haven't fully thought through or tested the idea though, just mentioning as "_maybe_ this is a scenario where you would set `webViewKey` but not `keepWebViewInstanceAfterUnmount`...."

Reply:
@TheAlmightyBob I think that would be a good use for the webViewKey. However, @donaldchen and I believe that for now it might just make sense to keep `webViewKey` alone and remove `keepWebViewInstanceAfterUnmount`. We can reintroduce that property at a later point when needed (perhaps, when supporting pop-up windows).

Reply:
It looks like Neal addressed this in https://github.com/react-native-webview/react-native-webview/pull/2469/commits/63c586af927c00e1005c754dc36e211b7f1907c3 , so I'm going to resolve this comment thread

Reply:
@donaldchen Er, does that mean you *don't* want me to resolve threads as I review changes...?